internal/SharedStateDecorator/SharedStateDecorator.js

```diff
@@ -40,107 +40,106 @@ const defaultConfig = {
 const SharedStateDecorator = hoc(defaultConfig, (config, Wrapped) => {
 	const {idProp, updateOnMount} = config;
 
-	return class extends Component {
-		static displayName = 'SharedStateDecorator';
-
-		static contextType = SharedState;
-
-		static propTypes = {
-			/**
-			 * Prevents the component from setting or restoring any framework shared state.
-			 *
-			 * @type {Boolean}
-			 * @default false
-			 * @public
-			 */
-			noSharedState: PropTypes.bool
-		};
-
-		constructor (props) {
-			super(props);
-
-			this.data = {};
-			this.sharedState = this.initSharedState();
-			this.state = {
-				updateOnMount: false
-			};
-		}
-
-		componentDidMount () {
-			this.loadFromContext();
-		}
-
-		componentDidUpdate (prevProps) {
-			if (!prevProps.noSharedState && this.props.noSharedState) {
-				this.data = {};
-			} else if (prevProps.noSharedState && !this.props.noSharedState) {
-				this.loadFromContext();
-			}
-		}
+	// eslint-disable-next-line no-shadow
+	const SharedStateDecorator = (props) => {
+		const context = useContext(SharedState);
+		const data = useRef({});
+		// eslint-disable-next-line no-unused-vars
+		const [updateOnMountState, setUpdateOnMountState] = useState(false);
```

Comment:
Since we don't use the `updateOnMountState`, we could write it like: `const [, setUpdateOnMountState] = useState(false);`